### PR TITLE
Heretic: Support 8 pages of saves

### PR DESCRIPTION
**Pull request by mikeday0**
Submitting the Heretic implementation of this for review. Once we get everything in a happy state I'll port it over to Hexen.

### CODE REVIEW

File: src/heretic/p_saveg.c [hunk 1]
```diff
@@ -47,7 +48,8 @@ char *SV_Filename(int slot)
     filename_len = strlen(savegamedir) + strlen(SAVEGAMENAME) + 8;
     filename = malloc(filename_len);
     M_snprintf(filename, filename_len,
-               "%s" SAVEGAMENAME "%d.hsg", savegamedir, slot);
+               "%s" SAVEGAMENAME "%d.hsg", savegamedir,
```
Comment (fabiangreffrath) [inline src/heretic/p_saveg.c:51]: So, the current 7th and 8th savegame will appear on page 2?

File: src/heretic/mn_menu.c [hunk 2]
```diff
@@ -794,6 +790,30 @@ static void DrawFilesMenu(void)
     players[consoleplayer].messageTics = 1;
 }
 
+// [crispy] support additional pages of savegames
+static void DrawSaveLoadBottomLine(Menu_t * menu)
```
Comment (turol) [inline src/heretic/mn_menu.c:794]: If `menu` is not modified in this function it should be `const`.

File: src/heretic/p_saveg.c (hunk 1 above)
Comment (mikeday0) [reply src/heretic/p_saveg.c:51]: Correct! This way no one has to rename any existing saves to get them to show up in the list.

**Review by fabiangreffrath — APPROVED** [review]

**Comment by fabiangreffrath** [comment]
Looks good, thank you! Would you like to add Hexen first or should I merge this now?

**Comment by mikeday0** [comment]
> Looks good, thank you! Would you like to add Hexen first or should I merge this now?

I would like to add the Hexen implementation to this PR if that's okay.

**Comment by fabiangreffrath** [comment]
Sure, please move on.

**Comment by fabiangreffrath** [comment]
> I change my mind - let's do Hexen as a separate PR. Ready to merge!

Fine with me, thanks!

**Pull request merged**